### DISCUSSION TITLE
Alter the REV metric so that the subdomains have the same aspect ratio as image?

### DISCUSSION THREAD

**Discussion by MaPhewJ** [PMEAL/porespy#578] [Q&A] [answered]
Cubic subdomains limit the max size of the subdomain that is used in the REV function due to the high aspect ratio of my image. If I could randomly generate volumes that have the same aspect ratio as my original image this would be useful. Any ideas? Many thanks in advance!

**Comment by jgostick** [comment, marked as answer]
Hi @MaPhewJ, I was on vacay last week, sorry for the delay.  Your proposal is interesting.  I'm pretty sure we could easily control the aspect ratio of the cube, but that would require altering the function, and I'm not totally convinced this is a universally desired behavior.  Have you thought about zooming your image to be cubic?  If you use the scipy.ndimage.zoom function you can zoom different amounts in different directions, so ``zoom(im, zoom=[1, 1, 10], order=0)`` would expand the z-dimension by a factor of 10.  I "think" the result would actually be valid "as is" since you'll get 10x more pixels in your volume, but your volume will be 10x larger, so this should cancel out.  Worth a shot maybe?

**Reply by MaPhewJ** [reply]
@jgostick , interesting suggestion thanks, I'll give it a go too. The work around I went with in the end was to crop the image into a series of rectangular volumes and then concatenate them together again in such a way that the volume was more cube like. 
I agree that its a slightly odd to not want cubic subdomains for a REV, but I work with thin sheet-like filter materials so that geometry was limiting here. I did find that in the end the REV was a few hundred voxels larger than a cube with side lengths equal to the shortest axis of the original image. So the work around was needed. 
Cheers again, really enjoying using porespy at the moment!

**Reply by jgostick** [reply]
> really enjoying using porespy at the moment!

Great to hear! If you end up implementing your aspect-ratio idea, perhaps you could consider contributing it :-)
